app/controllers/my/bookings_controller.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module My
+
+  class BookingsController < ApplicationController
+
+    before_action :authenticate_user!
+    before_action :find_booking, only: %i[edit update destroy]
+
+    def index
+      @bookings = current_user.bookings
+    end
+
+    def edit
+    end
+
+    def update
+      if @booking.update(booking_params)
```

Comment:
It does not allow booking extension.

Imagine you book from 2023-06-21 to 2023-06-22. You can not extend to 2023-06-25 because `Room#can_book?` does not check whom the booking belongs to. It just see that a booking already exists for those dates.